routes/artifacts.js

```diff
@@ -386,18 +387,18 @@ var replyWithArtifact = async function(taskId, runId, name, req, res) {
         if (this.artifactRegion === 'us-east-1') {
           vhost = 's3.amazonaws.com';
         } else {
-          vhost = `s3-${this.canonicalArtifactRegion}.amazonaws.com`;
+          vhost = `s3-${this.artifactRegion}.amazonaws.com`;
         }
 
         // Generate the public URL for this artifact
         var canonicalArtifactUrl = urllib.format({
           protocol: 'https:',
-          host: vhost,
+          host: bucket + '.' + vhost,
           pathname: prefix,
```

Comment:
All of this logic actually exists in `this.publicBucket.createGetUrl(prefix);`

I think we're better off using that... and then giving it an optional flag, like:
`this.publicBucket.createGetUrl(prefix, forceS3);`
So that if `forceS3` is true, then we don't use CDN (even if available).

Obviously, we default to `forceS3 = false`